### PR TITLE
Added `Miri` to `CI`

### PR DESCRIPTION
**Pull request by cospectrum**
I've separated all the tests and benchmarks where it makes sense. `benches` are ignored.

UB detected in `nalgebra` (`Svd::try_new`) and `rayon` (if feature is enabled), they are ignored.

### CODE REVIEW

File: src/region_labelling.rs [hunk 1]
```diff
@@ -275,7 +275,7 @@ mod tests {
 
     // One huge component with eight-way connectivity, loads of
     // isolated components with four-way connectivity.
-    fn chessboard(width: u32, height: u32) -> GrayImage {
+    pub(super) fn chessboard(width: u32, height: u32) -> GrayImage {
```
Comment (cospectrum) [inline src/region_labelling.rs:278]: It's used in `super::benches`

File: src/contrast.rs [hunk 2]
```diff
@@ -364,6 +364,7 @@ mod tests {
         }
     }
 
+    #[cfg_attr(miri, ignore)]
```
Comment (theotherphil) [inline src/contrast.rs:367]: Opting out for each benchmark individually feels a bit error prone - could you do the tests vs benches module split everywhere?

Comment (cospectrum) [reply src/contrast.rs:367]: Sure, just didn't want to scare you with code moves all over the place

**Comment by theotherphil** [comment]
@cospectrum as there’s been a burst of activity in this repo recently and your change touches a lot of files, let’s merge this after fixing the current merge conflict rather.

File: src/contrast.rs (hunk 2 above)
Comment (cospectrum) [reply src/contrast.rs]: Then let's do it in the second PR.

**Comment by cospectrum** [comment]
@theotherphil Conflict fixed.

**Comment by cospectrum** [comment]
Note: we want to ignore `benches` because they are slow, not because they don't pass the `Miri` check.

**Comment by cospectrum** [comment]
> @cospectrum is this good to go or do you have more changes pending for this PR?

You can merge

**Comment by theotherphil** [comment]
Thanks a lot for all the work you’re doing getting a proper testing suite in place.

**Pull request merged**